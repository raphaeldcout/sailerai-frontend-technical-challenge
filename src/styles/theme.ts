const theme = {
    colors: {
        primary: '#183145',
        primaryLight: '#788AA3',
        primaryDark: '#212121',
        primaryMedium: '#11283a',
        secondary: '#38b3ff',
        background: '#1f1f22',
        surface: '#f5f5f5',
        error: '#d32f2f',
        success: '#388e3c',
        successLight: '#2DD881',
        warning: '#fbc02d',
        border: '#e0e0e0',
        light: '#ffffff',
        neutralLowLight: '#b0d4ee',
        neutralLowMedium: '#f1f1f1',
        neutralLowDark: '#c4c4c4',
        neutralMediumLight: '#6c757d',
        neutralMedium: '#495057',
        greenLowLight: '#dcf6c6',
        rbg: {
            greenLowMedium: 'rgba(46, 178, 110, 0.2)',
            greenLowDark: 'rgba(46, 178, 110, 0.3)',
        }
    },
    darkMode: {
        foregroundRgb: '255, 255, 255',
        backgroundStartRgb: '0, 0, 0',
        backgroundEndRgb: '19, 19, 22',
    },
    spacings: {
        xs: '4px',
        sm: '8px',
        md: '16px',
        lg: '24px',
        xl: '32px',
        xxl: '64px',
    },
    typography: {
        fontSize: {
            xs: '12px',
            sm: '14px',
            md: '16px',
            lg: '20px',
            xl: '24px',
            xxl: '32px',
        },
        fontWeight: {
            light: 300,
            normal: 400,
            medium: 500,
            bold: 700,
        },
        lineHeight: {
            xs: '1rem',
            sm: '1.2rem',
            md: '1.4rem',
            lg: '1.6rem',
            xl: '1.8rem',
        },
    },
    borders: {
        radius: {
            none: '0px',
            sm: '4px',
            md: '8px',
            lg: '12px',
            full: '999px',
        },
        width: {
            thin: '1px',
            thick: '2px',
        },
    },
    shadows: {
        sm: '0px 1px 3px rgba(0, 0, 0, 0.12)',
        md: '0px 3px 6px rgba(0, 0, 0, 0.16)',
        lg: '0px 10px 20px rgba(0, 0, 0, 0.2)'
    },
    zIndex: {
        dropdown: 1000,
        modal: 1100,
        tooltip: 1200,
    },
    breakpoints: {
        xsmall: 575.98,
        small: 767.98,
        medium: 991.98,
        large: 1199.98,
        xlarge: 1399.98
    }
};

export const colors = theme.colors;

export type Theme = typeof theme;

export type ColorsType = keyof Theme['colors'];

export default theme;