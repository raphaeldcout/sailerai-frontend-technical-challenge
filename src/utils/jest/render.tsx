import { FC, ReactElement, ReactNode } from 'react';

import { ThemeProviderSailer } from '@/styles/withThemeProvider';
import { render, RenderOptions } from '@testing-library/react';

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <ThemeProviderSailer>{children}</ThemeProviderSailer>
);

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };

