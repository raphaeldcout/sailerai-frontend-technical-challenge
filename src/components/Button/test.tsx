import '@testing-library/jest-dom';

import { FiEdit, FiSend } from 'react-icons/fi';

import theme from '@/styles/theme';
import { render } from '@/utils/jest/render';
import { screen } from '@testing-library/react';

import { Button } from '.';

describe('Button', () => {
  test('should match snapshot only label', () => {
    const { container } = render(<Button label="Botão" loading />);

    expect(container).toMatchSnapshot();
  });

  test('should match snapshot with icons', () => {
    const { container } = render(
      <Button
        label="Botão"
        iconLeft={<FiEdit size={20} color={theme.colors.surface} />}
        iconRight={<FiSend size={20} color={theme.colors.surface} />}
      />
    );

    expect(container).toMatchSnapshot();
  });

  test('should show label', () => {
    render(<Button label="Botão" />);

    const label = screen.getByTestId('root-button-label');

    expect(label).toBeDefined();
  });

  test('should show icon left', () => {
    render(
      <Button
        iconLeft={<FiEdit size={20} color={theme.colors.surface} />}
        label="Botão"
      />
    );

    const element = screen.getByTestId('root-button-icon-left');

    expect(element).toBeDefined();
  });

  test('should not show icon left', () => {
    render(
      <Button
        iconLeft={undefined}
        label="Botão"
      />
    );

    const element = screen.queryByTestId('root-button-icon-left');

    expect(element).toBeNull();
  });

  test('should hide label when responsiveMode is enabled', () => {
    render(<Button responsiveMode label="Botão" />);

    const element = screen.queryByTestId('root-button-label');

    expect(element).toBeNull();
  });

  test('should show icon right', () => {
    render(
      <Button
        iconRight={<FiSend size={20} color={theme.colors.surface} />}
        label="Botão"
      />
    );

    const element = screen.getByTestId('root-button-icon-right');

    expect(element).toBeDefined();
  });

  test('should not show icon right', () => {
    render(
      <Button
        iconRight={undefined}
        label="Botão"
      />
    );

    const element = screen.queryByTestId('root-button-icon-right');

    expect(element).toBeNull();
  });

  test('should show loading behavior', () => {
    render(
      <Button
        loading
        label="Botão"
      />
    );

    const element = screen.getByTestId('root-button-loading');

    expect(element).toBeDefined();
  });

  test('should not show loading behavior', () => {
    render(<Button loading={false} label="Botão" />);

    const element = screen.queryByTestId('root-button-loading');

    expect(element).toBeNull();
  });
});
