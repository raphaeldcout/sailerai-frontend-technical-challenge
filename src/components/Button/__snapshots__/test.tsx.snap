// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot only label 1`] = `
<div>
  <button
    class="sc-dbef9764-0 hSiXXX"
    color="successLight"
    data-testid="root-button"
  >
    <p
      class="sc-dc096805-0 fZKqTq"
      color="surface"
      data-testid="root-button-label"
      variant="medium"
      weight="medium"
    >
      Botão
    </p>
    <span
      class="sc-dbef9764-1 gOepMG"
      data-testid="root-button-loading"
    >
      <svg
        class="sc-dbef9764-2 jAnwkV"
        color="#f5f5f5"
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        style="color: rgb(245, 245, 245);"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="23 4 23 10 17 10"
        />
        <path
          d="M20.49 15a9 9 0 1 1-2.12-9.36L23 10"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`Button should match snapshot with icons 1`] = `
<div>
  <button
    class="sc-dbef9764-0 hSiXXX"
    color="successLight"
    data-testid="root-button"
  >
    <span
      class="sc-dbef9764-1 gOepMG"
      data-testid="root-button-icon-left"
    >
      <svg
        color="#f5f5f5"
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        style="color: rgb(245, 245, 245);"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
        />
        <path
          d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
        />
      </svg>
    </span>
    <p
      class="sc-dc096805-0 fZKqTq"
      color="surface"
      data-testid="root-button-label"
      variant="medium"
      weight="medium"
    >
      Botão
    </p>
    <span
      class="sc-dbef9764-1 gOepMG"
      data-testid="root-button-icon-right"
    >
      <svg
        color="#f5f5f5"
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        style="color: rgb(245, 245, 245);"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="22"
          x2="11"
          y1="2"
          y2="13"
        />
        <polygon
          points="22 2 15 22 11 13 2 9 22 2"
        />
      </svg>
    </span>
  </button>
</div>
`;
